lib/iris/cube.py

```diff
@@ -2785,6 +2785,34 @@ def slices(self, ref_to_slice, ordered=True):
 
         return _SliceIterator(self, dims_index, dim_to_slice, ordered)
 
+    def invert_dims(self, dimensions):
+        """
+        Invert the specified dimensions of a cube in-place.
+
+        Args:
+
+        * dimensions (iterable of ints):
```

Comment:
Would it be useful to be able to specify the coordinate by name or coordinate instance here in stead of number? If we imagine a future time when inversion of dimensions with AuxCoords spanning them is implemented, this may be very useful. Since this affects the public API it should be ironed out now (the actual implementation for AuxCoords/multi-dimensional coords can indeed be added at a later time as the need arises).

I don't think this would change the implementation much at all. We already have a way of accepting "string or instance" as input which you could reuse. Then you just proceed as before, check if the dimension spanned by the coord has other coordinates or if it spans multiple dimensions and reject if so.